### PR TITLE
fix(paymaster-example): refactor functions to make it closer to the tutorial

### PR DESCRIPTION
**Pull request by jakim929**
the behavior should be identical, this just makes it easier to explain what is going on, and closer to the tutorial readmes.

### CODE REVIEW

**Comment by jakim929** [comment]
* **#170** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/ecosystem/170?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jakim929 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/paymaster-nft-mint/src/libraries/aa-sdk/createModularAccountClientWithPaymaster.ts [hunk 1]
```diff
@@ -1,29 +1,61 @@
...
+    chain,
+  })
+
+  const modularAccount = await createMultiOwnerModularAccount({
+    signer: signer,
+    chain: chain,
+    transport: http(rpcUrl),
+  })
+
+  return createSmartAccountClient({
```
Comment (jakim929) [inline apps/paymaster-nft-mint/src/libraries/aa-sdk/createModularAccountClientWithPaymaster.ts:37]: this is easier to grok than the separate `superchainPaymasterMiddleware` function we had previously

File: apps/paymaster-nft-mint/src/libraries/permissionless/KernelPermissionlessCard.tsx [hunk 2]
```diff
@@ -40,10 +39,9 @@ const useSendMintNft = ({
       }
 
       const transactionHash = await smartAccountClient.writeContract({
-        abi: SimpleNftAbi,
+        abi: [parseAbiItem('function mint() returns (uint256)')],
```
[the remark this answers is not in the document]
Comment (tremarkley) [reply apps/paymaster-nft-mint/src/libraries/permissionless/KernelPermissionlessCard.tsx:42]: interesting, so this no longer needs to specify which address to mint to?

**Review by tremarkley — APPROVED** [review]

File: apps/paymaster-nft-mint/src/libraries/permissionless/KernelPermissionlessCard.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (jakim929) [reply apps/paymaster-nft-mint/src/libraries/permissionless/KernelPermissionlessCard.tsx:42]: the contract supports both! https://optimism-sepolia.blockscout.com/address/0xAB559628B94Fd9748658c46E58a85EfB52FdaCa6?tab=write_contract

**Pull request merged**